Feature/user profile

tpan501 opened the pull request:
[try to fix the scrollbar problem by adding a constant (60) and replace the fixed parameter in tailwindcss config file.

simplify the route names and replace many HTML elements with components

optimize the display of the table.

Haven't changed the display of adding tags in account setting, need other people to decide.

tpan501 committed:
create Userprofile page and its sub-component (e.g. public profile, my group, apply in progress, account settings etc.) Added routes for the userprofilepage.
everything is static now, haven't connected with DB

tpan501 committed:
fix adding tags bug and optimize the function

tpan501 committed:
fix a few bugs in the frontend related to routes
and CSS styles

tpan501 committed:
try to fix the scrollbar problem by adding a constant (60) and replace the fixed parameter in tailwindcss config file.
simplify the route names and replace many HTML elements with components.

tpan501 committed:
optimize the tags deletion in account settings

jlmjkfd commented on frontend/src/components/SideBarOption.jsx:
This component is only used in UserPageSideBar, I think you may just write in that file, and not necessary to separate it in a new file. If you really want to separate it, consider renaming it to 'UserPageSideBarOption.jsx' so that these two files would display together in the folder, which makes it easier to reflect the relationship between these two files.

tpan501 replied:
This has been done

jlmjkfd commented on frontend/src/components/ProfileTags.jsx:
This component will be replaced by a common Tag component in the future as we discussed today. You may write it in PublicProfilePage.jsx temporarily. This file is not necessary.

tpan501 replied:
This has been done. thanks.

jlmjkfd commented on frontend/src/pages/AccountSettingsPage.jsx:
```diff
@@ -0,0 +1,199 @@
+                                disabled={!isEditing}
+                            />
+                            Other
+                        </label>
```
Remove other? I think it's weird.

tpan501 replied:
OK

jlmjkfd commented on frontend/src/pages/UserPage.jsx:
```diff
@@ -0,0 +1,25 @@
+    return (
+        <div className="overflow-y-auto">
+            <Navbar />
+            <div className="flex overflow-y-auto" style={{ minHeight: contentHeight }}>
```
Not necessary to introduce a constant in such a complex way. Write "min-h-[calc(100vh-theme('spacing.navHeight'))" in className and you can roll everything back. One thing to note: tailwind cannot handle the underscore character in this case. I have asked Ryan to change nav_height to navHeight in tailwind.config.js. You can modify it to try this code.

tpan501 replied:
Thanks for the solution but after merging the navbar I found the original css is not applicable, however, the code below solves the problem as well,  please double check for me in my next commit.
```
const UserPage = () => {
    return (
        <div className="overflow-y-auto">
            <Navbar />
            <div className="flex pt-10 overflow-y-auto min-h-screen" > {/*I didn't add calc！*/}
                <UserPageSideBar />
                <div className="flex-grow" >
                    <Outlet />
                </div>
            </div>
        </div>
    );
};
```

jlmjkfd replied:
It woks.

jlmjkfd replied:
I found that this works because the nav bar is set fixed. So it masks the top 60px of your page. Be aware of this. You may need to set an additional 60px of the margin-top for user pages.

tpan501 replied:
so clever are you. I do set additional top padding for other components...
```JSX
<div className="flex pt-10 overflow-y-auto min-h-screen" >
                <UserPageSideBar />
                <div className="flex-grow" >
                    <Outlet />
                </div>
            </div>
```

jlmjkfd replied:
I asked Ryan to add a placeholder behind the nav so that we don't need to add padding on every page. You may delete the padding later on.

tpan501 replied:
will do

jlmjkfd commented on frontend/src/utils/constants.js:
Not necessary

jlmjkfd commented on frontend/tailwind.config.js:
```diff
@@ -9,7 +10,7 @@ export default {
       spacing: {
         main_content: "1280px",
-        nav_height: '60px',
+        nav_height: `${NAV_HEIGHT}px`,
```
navHeight: '60px'
Delete line 2.

tpan501 replied:
Done. Also I wonder if we need to change "main_content" to "mainContent"?

jlmjkfd replied:
Yes, we need to avoid all the underscores in this config file. Make sure you change it everywhere in classNames.

tpan501 replied:
will do.

jlmjkfd reviewed (commented):
Out of Alignment in my computer
![image](https://github.com/UOA-CS732-SE750-Students-2024/project-group-noble-narwhals/assets/6456953/8c02a46c-78a9-48d1-82ee-820971388219)

tpan501 committed:
Merge remote-tracking branch 'origin/develop' into feature/userProfile
Also, integrate some components to reduce the component number.
Adjust the CSS, please check the style if it is working, if something doesn't work on your computer please let me know

tpan501 committed:
in case of conflict i think i need to change it back to its original version

ryanwooa commented on frontend/tailwind.config.js:
```diff
@@ -1,4 +1,5 @@
 /** @type {import('tailwindcss').Config} */
+const { NAV_HEIGHT } = require('./src/utils/constants');
```
Delete this line

ryanwooa commented:
![image](https://github.com/UOA-CS732-SE750-Students-2024/project-group-noble-narwhals/assets/119087910/e173ec1d-57a6-4816-84e7-23de89b92dc8)
- the gap between content and sidebar is different in different page, should be the same
- the font style of title of sections in different page is not same. Considering make them same
- How about write the emaill under the username, in public profile page

jlmjkfd commented on frontend/src/components/UserPageSideBar.jsx:
```diff
@@ -0,0 +1,72 @@
+        onClick={() => handleClick(label)}
+      >
+        <Link to={to} className="flex items-center w-full justify-start pl-4">
+          {icon}
```
Please set all the icons to the same size using CSS.

tpan501 committed:
delete the "require (.src/utils/constant.js)

tpan501 committed:
this commit covers the changes before merging with the new router arrangement.
-fixed the sidebar width bug
-relocate the email information
-optimize the structure

tpan501 committed:
Merge remote-tracking branch 'origin/develop' into feature/userProfile

NOTE: I comment out the original page components because they are not used anymore. If you think they can be deleted, I will do that in the next commit.

tpan501 committed:
adding more padding and margin from 4 to 5 to make the alignment in setting page the same as other pages.

zli213 commented on frontend/src/App.jsx:
```diff
@@ -2,6 +2,12 @@ import React from "react";
+// import PublicProfilePage from "./pages/PublicProfilePage";
+// import AccountSettingsPage from "./pages/AccountSettingsPage";
+// import ApplyInProgressPage from "./pages/ApplyInProgressPage";
+// import LikedGroupsPage from "./pages/LikedGroupsPage";
```
Delete these comments.

zli213 commented on frontend/src/components/GroupTableRow.jsx:
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+
+export function GroupTableRow({ groupName, category, due, groupOwner, members, status }) {
```
It is suggested to simplify it by using a single attribute named "properties" instead, and then extract it as written below：
```
const groupName = properties.groupName;
const category = properties.category;
```

zli213 commented on frontend/src/components/UserPageSideBar.jsx:
```diff
@@ -0,0 +1,73 @@
+import { Link } from 'react-router-dom';
+import { CgProfile } from "react-icons/cg";
+import { IoSettingsSharp } from "react-icons/io5";
+// import { FaUserGroup } from "react-icons/fa6";
```
If u don't use it, please delete it

zli213 commented on frontend/src/pages/AccountSettingsPage.jsx:
Why u comment all but upload it?

tpan501 replied:
I've written in the commit message saying it is for my reviewers to double check if they are okay to deleted and if yes I will do that in the next commitment

zli213 commented on frontend/src/pages/user/AccountSettingPage.jsx:
```diff
@@ -1,3 +1,188 @@
+    const [gender, setGender] = useState("Male");
+    const [email, setEmail] = useState("InitialEmail");
+    const [password, setPassword] = useState("InitialPassword");
+    const [tags, setTags] = useState(["exampleTag1", "exampleTag2"]); // 假设这是初始的tags
```
Please translate to English.

zli213 reviewed (commented):
I personally feel that the content marked in red should be hidden before a user clicks to modify their profile. Additionally, it might be worth considering arranging the delete button alongside the modify button to save space.
![CleanShot 2024-03-28 at 11 40 06@2x](https://github.com/UOA-CS732-SE750-Students-2024/project-group-noble-narwhals/assets/127060502/bc893d9d-0fd6-473d-8a3e-d23e3f67aa69)

xinranwangxwan559 commented on frontend/src/pages/AccountSettingsPage.jsx:
I think we need to use English for comment

xinranwangxwan559 commented on frontend/src/pages/PublicProfilePage.jsx:
Will u delete this file? These comments are redundant.

xinranwangxwan559 commented on frontend/src/pages/UserPage.jsx:
same question, these comments are redundant..

tpan501 replied:
yes I have understood this from Ralph's comment and will do this soon

tpan501 committed:
delete the original page components because of merging from the new router arrangment

translate some comment into english.

delete some unused code.

tpan501 committed:
optimize the parameter of Grouptablerow component by replacing it by "props"

tpan501 commented:
> I personally feel that the content marked in red should be hidden before a user clicks to modify their profile. Additionally, it might be worth considering arranging the delete button alongside the modify button to save space. ![CleanShot 2024-03-28 at 11 40 06@2x](https://private-user-images.githubusercontent.com/127060502/317537470-bc893d9d-0fd6-473d-8a3e-d23e3f67aa69.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE1ODI0NTMsIm5iZiI6MTcxMTU4MjE1MywicGF0aCI6Ii8xMjcwNjA1MDIvMzE3NTM3NDcwLWJjODkzZDlkLTBmZDYtNDczZC04YTNlLWQyM2UzZjY3YWE2OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyN1QyMzI5MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMzA1NDllNGRkMTA5ZWE2ZjQ3ZDU3MDdhMWNiMzY3N2VjMmE1ZWEwYWRlZWM0MDllZDZmZTlmYWI0NWY0ODg2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.pgh_HSPo6s26PZUeVBaJOaTXZGrYI8LEVKxsFuQnAzI)

Put Delete account button at the bottom of the page is for avoiding user from mis-clicking it. What do you think?
For hiding the components when isEditing is false, will consider about it.

tpan501 committed:
modify the setting page by hiding some components when isEditing is false

tpan501 committed:
update the userprofile's display format from table to "long block" according to the styling of search results

tpan501 committed:
delete the grouptable component because we won't use table any more

tpan501 committed:
Merge branch 'develop' into feature/userProfile
